pkg/bpf/map_linux.go

```diff
@@ -245,11 +257,51 @@ func (m *Map) scheduleErrorResolver() {
 // user space in a local cache (map) and will indicate the status of each
 // individual entry.
 func (m *Map) WithCache() *Map {
-	m.cache = map[string]*cacheEntry{}
+	if m.cache == nil {
+		m.cache = map[string]*cacheEntry{}
+	}
+	m.withCache = true
 	m.enableSync = true
 	return m
 }
 
+// WithPressureMetricThreshold enables the tracking of a metric that measures
+// the pressure of this map. This metric is only reported if over the
+// threshold.
+func (m *Map) WithPressureMetricThreshold(threshold float64) *Map {
+	// When pressure metric is enabled, we keep track of map keys in cache
+	if m.cache == nil {
+		m.cache = map[string]*cacheEntry{}
+	}
```

Comment:
Maybe would be better to call `m.WithCache()` here? otherwise we will have `m.cache` with data but `m.withCache` will be false.

Reply:
`m.cache` and `m.WithCache()`/`m.withCache` are different things. `m.cache` is used also for the pressure metric key-only cache unless `m.withCache` flag is set in which case is a key-value cache.  Perhaps renaming `m.withCache` to `m.withValueCache` is a bit more clear.